hivemq-edge/src/frontend/src/extensions/datahub/components/nodes/NodeWrapper.tsx

```diff
@@ -1,19 +1,29 @@
-import { FC, ReactNode, useEffect, useState } from 'react'
+import { FC, ReactNode, useEffect, useMemo, useState } from 'react'
 import { useLocation, useNavigate } from 'react-router-dom'
 import { NodeProps } from 'reactflow'
-import { BoxProps, Card, CardBody, CardBodyProps, HStack } from '@chakra-ui/react'
+import { Avatar, BoxProps, Card, CardBody, CardBodyProps, HStack } from '@chakra-ui/react'
+import { RiPassExpiredLine, RiPassPendingLine, RiPassValidLine } from 'react-icons/ri'
 
-interface NodeWrapperProps extends NodeProps {
+import { DataHubNodeData, PolicyDryRunStatus } from '@datahub/types.ts'
+
+interface NodeWrapperProps extends NodeProps<DataHubNodeData> {
   children: ReactNode
   route: string
   wrapperProps?: CardBodyProps
 }
 
-export const NodeWrapper: FC<NodeWrapperProps> = ({ selected, children, route, wrapperProps }) => {
+export const NodeWrapper: FC<NodeWrapperProps> = ({ selected, children, route, wrapperProps, data }) => {
   const [internalSelection, setInternalSelection] = useState(false)
   const navigate = useNavigate()
   const { pathname } = useLocation()
 
+  const CheckIcon = useMemo(() => {
+    if (data.dryRunStatus === PolicyDryRunStatus.IDLE) return RiPassPendingLine
+    if (data.dryRunStatus === PolicyDryRunStatus.SUCCESS) return RiPassValidLine
+    if (data.dryRunStatus === PolicyDryRunStatus.FAILURE) return RiPassExpiredLine
+    return RiPassPendingLine
```

Comment:
suggestion: extract this value into a util function since it's defined twice

https://github.com/hivemq/hivemq-edge/pull/288/files#diff-b402264e2a1a394a7851c7b8bd0bf5b707a7cb543d10e584ed93d2c5ab931587R91-R95